scripts/gen-desktop-files

```diff
@@ -0,0 +1,517 @@
+        ;;
+      \'blackarch-gpu\')
+        categorie="${categorie} X-Blackarch-Gpu;"
+        ;;
+      \'blackarch-hardware\')
+        categorie="${categorie} X-Blackarch-Hardware;"
+        ;;
+      \'blackarch-honeypot\')
+        categorie="${categorie} X-Blackarch-Honeypot;"
+        ;;
+      \'blackarch-ids\')
+        categorie="${categorie} X-Blackarch-Ids;"
+        ;;
+      \'blackarch-keylogger\')
+        categorie="${categorie} X-Blackarch-Keylogger;"
+        ;;
+      \'blackarch-malware\')
+        categorie="${categorie} X-Blackarch-Malware;"
+        ;;
+      \'blackarch-misc\')
+        categorie="${categorie} X-Blackarch-Misc;"
+        ;;
+      \'blackarch-mobile\')
+        categorie="${categorie} X-Blackarch-Mobile;"
+        ;;
+      \'blackarch-mobile-reversing\')
+        categorie="${categorie} X-Blackarch-Mobile-Reversing;"
+        ;;
+      \'blackarch-networking\')
+        categorie="${categorie} X-Blackarch-Networking;"
+        ;;
+      \'blackarch-nfc\')
+        categorie="${categorie} X-Blackarch-Nfc;"
+        ;;
+      \'blackarch-packer\')
+        categorie="${categorie} X-Blackarch-Packer;"
+        ;;
+      \'blackarch-proxy\')
+        categorie="${categorie} X-Blackarch-Proxy;"
+        ;;
+      \'blackarch-radio\')
+        categorie="${categorie} X-Blackarch-Radio;"
+        ;;
+      \'blackarch-recon\')
+        categorie="${categorie} X-Blackarch-Recon;"
+        ;;
+      \'blackarch-reversing\')
+        categorie="${categorie} X-Blackarch-Reversing;"
+        ;;
+      \'blackarch-scan\')
+        categorie="${categorie} X-Blackarch-Scan;"
+        ;;
+      \'blackarch-scanner\')
+        categorie="${categorie} X-Blackarch-Scanner;"
+        ;;
+      \'blackarch-sniffer\')
+        categorie="${categorie} X-Blackarch-Sniffer;"
+        ;;
+      \'blackarch-social\')
+        categorie="${categorie} X-Blackarch-Social;"
+        ;;
+      \'blackarch-spoof\')
+        categorie="${categorie} X-Blackarch-Spoof;"
+        ;;
+      \'blackarch-stego\')
+        categorie="${categorie} X-Blackarch-Stego;"
+        ;;
+      \'blackarch-tunnel\')
+        categorie="${categorie} X-Blackarch-Tunnel;"
+        ;;
+      \'blackarch-unpacker\')
+        categorie="${categorie} X-Blackarch-Unpacker;"
+        ;;
+      \'blackarch-voip\')
+        categorie="${categorie} X-Blackarch-Voip;"
+        ;;
+      \'blackarch-webapp\')
+        categorie="${categorie} X-Blackarch-Webapp;"
+        ;;
+      \'blackarch-windows\')
+        categorie="${categorie} X-Blackarch-Windows;"
+        ;;
+      \'blackarch-wireless\')
+        categorie="${categorie} X-Blackarch-Wireless;"
+        ;;
+    esac
+  done
+  echo "$categorie"
+}
+
+
+# Add .desktop file to sources
+#
+# Arguments:
+#   pkg: PKGBUILD file
+#   name: desktop/icon file name
+#
+add_source()
+{
+  pkg="$1"
+
+  # if no name is provided, add ba-$pkgname to PKGBUILD file
+  if [ -z "$2" ]; then
+    name='ba-$pkgname.desktop'
```

Comment:
Expressions don't expand in single quotes, use double quotes for that.
```suggestion
    name="ba-$pkgname.desktop"
```

Reply:
It's the intention, that it is not expanded here, but used in PKGBUILD file

Reply:
Ok, please review the other suggestions as well.